crates/driver/src/tests/setup/mod.rs

```diff
@@ -1224,6 +1225,45 @@ impl<'a> SolveOk<'a> {
         assert!(self.solutions().is_empty());
     }
 
+    /// Check that the solution contains the expected JIT orders.
+    pub fn jit_orders(self, jit_orders: &[JitOrder]) -> Self {
+        let solution = self.solution();
+        assert!(solution.get("orders").is_some());
+        let trades = serde_json::from_value::<HashMap<String, serde_json::Value>>(
+            solution.get("orders").unwrap().clone(),
+        )
+        .unwrap();
+
+        // Since JIT orders don't have UID at creation time, we need to search for
+        // matching token pair
+        for expected in jit_orders.iter() {
+            let mut exist = false;
+            for trade in trades.values() {
+                exist |= self.exist_jit_order(trade, expected)
+            }
+            assert!(exist, "JIT order {expected:?} not found");
```

Comment:
nit: the construction seems too complex. Why don't you just:

```
for expected in jit_orders.iter() {
   for trade in trades.values() {
      let exist = self.exist_jit_order(trade, expected);
      assert!(exist, "JIT order {expected:?} not found");
   }
}
```

Reply:
Why not:
```rust
        for expected in jit_orders.iter() {
            let exists = trades.values().any(|trade| self.trade_matches(trade, expected));
            assert!(exists, "JIT order {expected:?} not found");
        }
```